src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -135,11 +137,23 @@ std::shared_ptr<StorageObjectStorageSource::IIterator> StorageObjectStorageSourc
     std::unique_ptr<IIterator> iterator;
     if (configuration->isPathWithGlobs())
     {
-        /// Iterate through disclosed globs and make a source for each file
-        iterator = std::make_unique<GlobIterator>(
-            object_storage, configuration, predicate, virtual_columns,
-            local_context, is_archive ? nullptr : read_keys, query_settings.list_object_keys_size,
-            query_settings.throw_on_zero_files_match, file_progress_callback);
+        auto path = configuration->getPath();
+        if (containsOnlyEnumGlobs(path))
```

Comment:
Why?
We can expand enum globs regardless of whether there are other globs. It is OK to expand e.g. `prefix{1,2,3}/file*.csv`

Reply:
Also, I think it is even OK to expand stuff like `prefix{1?,2?,3?}/file*.csv`. This is still better than matching it against regex